### ISSUE TITLE
Feature Request: When searching in multiple files, highlight the search results

### ISSUE THREAD

**Issue by lzh1043060917** [geany/geany#3380]
just like this，this software is notepad++：
![image](https://user-images.githubusercontent.com/25424316/215050347-b028ce73-de78-4397-8ae6-ca40dcc8343e.png)
but now in geany。When searching in multiple files, the search string is not highlighted in the search results, which makes it difficult for people to quickly see the search string.
![image](https://user-images.githubusercontent.com/25424316/215050920-244656c8-32e3-47b3-8afa-68509adf9fc7.png)

**Comment by elextr** [comment]
Geany does not search the files itself, instead it uses the `grep` program (as the blue line at the start of the results shows) which has been optimised for decades to search lots of files quickly.  The output of grep is shown in the results pane.  Unfortunately grep does not provide any indication of which characters in the output matched, so there is no way for Geany to highlight them.

**Comment by dolik-rce** [comment]
Actually, `grep --color=always` gives you exactly the information that would be needed for this. The matched characters would be surrounded by `\033[0;31m` and `\033[0m`. But relying on this would be hacky at best and could easily break with various grep implementations.

**Comment by elextr** [comment]
> But relying on this would be hacky at best and could easily break with various grep implementations.

Yes, thought of this before replying, but `--color=` is not [POSIX grep](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html).  Also the grep command is configurable, it might not even be grep, although it will need to accept the same basic options as grep does.

**Comment by lzh1043060917** [comment]
> Actually, gives you exactly the information that would be needed for this. The matched characters would be surrounded by and . But relying on this would be hacky at best and could easily break with various grep implementations.`grep --color=always``\033[0;31m``\033[0m`

How do I configure additional parameters when using Search in Folders? My parameter will be garbled when searching
![Uploading image.png…]()

**Comment by lzh1043060917** [comment]
> Actually, `grep --color=always` gives you exactly the information that would be needed for this. The matched characters would be surrounded by `\033[0;31m` and `\033[0m`. But relying on this would be hacky at best and could easily break with various grep implementations.

![image](https://user-images.githubusercontent.com/25424316/216778425-a13b7b9f-2f31-49de-83fc-75965bee273c.png)
How do I configure additional parameters when using Search in Folders? My parameter will be garbled when searching.

**Comment by elextr** [comment]
@lzh1043060917  you have used the right method, although I don't know what the `'`s are for?

But adding the option to grep almost certainly won't help, the compiler widget is not a terminal, it is unlikely to interpret escape sequences.  AFAIK "somebody" needs to add code to Geany to recognise the escape sequences and convert to Pango markup which should be recognised by the widget.
